File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/code/enums/CodeMode.java
package com.glowxq.oj.code.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum CodeMode implements BaseType {

    Glowc("Glowc", "Glowc"),

    OJ("OJ", "OJ");

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static CodeMode matchCode(String code) {
        for (CodeMode pushStatus : CodeMode.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/code/CodeMode.ts
import type { BaseEnum } from '@/enums/base';

interface CodeModeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

export const CodeMode: { [key: string]: CodeModeEnum } = {
  GlowC: {
    code: 'Glowc',
    name: 'Glowc模式',
    tooltip: 'Glowc代码模式',
    text: 'Glowc代码模式'
  },
  OJ: {
    code: 'OJ',
    name: 'OJ模式',
    tooltip: 'OJ代码模式',
    text: 'OJ代码模式'
  }
};

export function matchCode(code: string): CodeModeEnum | null {
  return Object.values(CodeMode).find(item => item.code === code) || null;
}
